src/OSPSuite.Core/Domain/Services/SpatialStructureMerger.cs

```diff
@@ -67,9 +67,9 @@ private void createMergedContainerStructureInRoot(IContainer root, SimulationBui
 
          //Merge all other spatial structures
          //make sure we map the container to a model container so that we do not change the original containers
-
-         allOtherSpatialStructuresWithMergeBehavior.Select(x => new {x.mergeBehavior, topContainers = x.spatialStructure.TopContainers.Select(mapToModelContainer).ToList()})
-            .Each(x => x.topContainers.Each(topContainer => mergeTopContainerInStructure(topContainer, root, x.mergeBehavior)));
+         allOtherSpatialStructuresWithMergeBehavior
+             .Select(item => new { item.mergeBehavior, topContainers = item.spatialStructure.TopContainers.Select(mapToModelContainer).ToList(), item })
```

Comment:
Could this be `item.spatialStructure` at the end instead of `item`. 

It's used in `tryMergeTopContainerInStructure` only to access the SpatialStructure (and the module from there).